services/src/main/java/org/keycloak/services/resources/admin/RealmAdminResource.java

```diff
@@ -966,6 +981,43 @@ public Response testLDAPConnection(TestLdapConnectionRepresentation config) {
                 config.getStartTls());
     }
 
+    /**
+     * Get LDAP supported extensions.
+     * @param config LDAP configuration
+     * @return
+     */
+    @POST
+    @Path("ldap-supported-extensions")
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(javax.ws.rs.core.MediaType.APPLICATION_JSON)
+    public Response getLDAPSupportedExtensions(TestLdapConnectionRepresentation config) {
+
+        auth.realm().requireManageRealm();
+
+        String bindCredential = handleSavedBindCredential(config.getComponentId(), config.getBindCredential());
+
+        // Create Factory Methods for LDAPConfig and LDAPIdentityStore in LDAPConnectionTestManager?
+        MultivaluedHashMap<String, String> map = new MultivaluedHashMap<>();
+        map.add(LDAPConstants.CONNECTION_URL, config.getConnectionUrl());
+        map.add(LDAPConstants.USE_TRUSTSTORE_SPI, config.getUseTruststoreSpi());
+        map.add(LDAPConstants.AUTH_TYPE, LDAPConstants.AUTH_TYPE_SIMPLE);
```

Comment:
That is because` AUTH_TYPE` is not part of `org.keycloak.representations.idm.TestLdapConnectionRepresentation` as you said. I think that the rootDSL query should be done using the same credentials then used later on, to avoid permission problems. So it might be unnecessary to pass AUTH_TYPE, unless something like SASL is added?

Reply:
Yes, I think that it may be needed to either:
1) add authType as another option to TestLdapConnectionRepresentation.
2) assume "bindType" none in case that bindDn and bindCredential are not set?

It seems that (2) might be easier? However the bindType none should be supported. In many cases, administrator sets the bindType "none" when configuring their LDAP provider. In this case, querying the LDAP server rootDSL should be also possible with the bindType "none" .

Reply:
Done. I added the authType to TestLdapConnectionRepresentation.